src/algebra/algebra/subalgebra.lean

```diff
@@ -743,6 +744,81 @@ set_like.coe_injective set.prod_inter_prod
 
 end prod
 
+section supr_lift
+variables {ι : Type*}
+
+lemma coe_supr_of_directed [nonempty ι] {S : ι → subalgebra R A}
+  (dir : directed (≤) S) : ↑(supr S) = ⋃ i, (S i : set A) :=
+let K : subalgebra R A :=
+  { carrier := ⋃ i, (S i),
+    mul_mem' := λ x y hx hy,
+      let ⟨i, hi⟩ := set.mem_Union.1 hx in
+      let ⟨j, hj⟩ := set.mem_Union.1 hy in
+      let ⟨k, hik, hjk⟩ := dir i j in
+      set.mem_Union.2 ⟨k, subalgebra.mul_mem (S k) (hik hi) (hjk hj)⟩ ,
+    add_mem' := λ x y hx hy,
+      let ⟨i, hi⟩ := set.mem_Union.1 hx in
+      let ⟨j, hj⟩ := set.mem_Union.1 hy in
+      let ⟨k, hik, hjk⟩ := dir i j in
+      set.mem_Union.2 ⟨k, subalgebra.add_mem (S k) (hik hi) (hjk hj)⟩,
+    algebra_map_mem' := λ r, let i := @nonempty.some ι infer_instance in
+      set.mem_Union.2 ⟨i, subalgebra.algebra_map_mem _ _⟩ } in
+have supr S = K,
+  from le_antisymm (supr_le (λ i, set.subset_Union (λ i, ↑(S i)) i))
+    (set_like.coe_subset_coe.1
+      (set.Union_subset (λ i, set_like.coe_subset_coe.2 (le_supr _ _)))),
+this.symm ▸ rfl
+
+/-- Define an algebra homomorphism on a directed supremum of subalgebras by defining
+it on each subalgebra, and proving that it agrees on the intersection of subalgebras. -/
+noncomputable def supr_lift [nonempty ι]
+  (K : ι → subalgebra R A)
+  (dir : directed (≤) K)
+  (f : Π i, K i →ₐ[R] B)
+  (hf : ∀ (i j : ι) (h : K i ≤ K j), f i = (f j).comp (inclusion h)) :
+  ↥(supr K) →ₐ[R] B :=
```

Comment:
I think it might pay of to have a `T = supr K` predicate here as well. What do you think?

Reply:
I've made this change now